packages/namekit-react/src/components/Identity.tsx

```diff
@@ -268,40 +293,69 @@ const Followers = ({ className, ...props }: SubComponentProps) => {
   );
 };
 
-interface ProfileLinkProps extends SubComponentProps {
-  config?: ProfileLinkConfig;
+interface ProfileLinkProps {
+  config?: ProfileLinkGenerator;
+  className?: string;
+  children?: React.ReactNode;
+  onClick?: (e: React.MouseEvent) => void;
 }
 
-const ProfileLink = ({
-  className,
+const ProfileLink: React.FC<ProfileLinkProps> = ({
+  config,
   children,
-  config: instanceConfig,
-  ...props
-}: ProfileLinkProps) => {
-  const { loadingState, address } = useIdentity();
-  const globalConfig = useNameKitConfig();
+  onClick,
+}) => {
+  const identity = useIdentity();
+  const nameKitConfig = useNameKitConfig();
 
-  if (loadingState !== "success") {
+  const linkConfig =
+    config || nameKitConfig.profileLinks?.[0] || DEFAULT_PROFILE_LINKS[0];
+
+  if (!identity) {
+    console.warn("ProfileLink used outside of Identity context");
     return null;
   }
 
-  const config = instanceConfig ||
-    globalConfig.profileLinks || {
-      getProfileURL: (address) => `https://app.ens.domains/${address}`,
-      getProfileLink: (address, children) => (
-        <a
-          href={`https://app.ens.domains/${address}`}
-          target="_blank"
-          rel="noopener noreferrer"
-          className={cc(["namekit-profile-link", className])}
-          {...props}
-        >
-          {children}
-        </a>
-      ),
-    };
+  const url = linkConfig.getProfileURL(identity.address);
 
-  return config.getProfileLink(address, children);
+  return (
+    <a
+      href={url}
+      target="_blank"
+      rel="noopener noreferrer"
+      className="namekit-profile-link"
+      onClick={onClick}
+    >
+      {children || linkConfig.getName()}
+    </a>
+  );
+};
+
+interface ProfileLinksProps {
```

Comment:
Let's have a chat to help align on goals here. I think we're drifting away from solving the problems that people are actually looking to solve.